preview/preview.cpp

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2020, Raspberry Pi (Trading) Ltd.
+ *
+ * preview.cpp - preview window interface
+ */
+
+#include "preview.hpp"
+
+Preview *make_null_preview(Options const *options);
+Preview *make_egl_preview(Options const *options);
+Preview *make_drm_preview(Options const *options);
+Preview *make_qt_preview(Options const *options);
+
+Preview *make_preview(Options const *options)
+{
+	if (options->nopreview)
+		return make_null_preview(options);
+	else if (options->qt_preview)
+		return make_qt_preview(options);
```

Comment:
Does this need an ``#if QT_PRESET`` type guard?

Reply:
Yes!